src/Neo.VM/Utility.cs

```diff
@@ -27,6 +27,23 @@ static Utility()
             StrictUTF8.EncoderFallback = EncoderFallback.ExceptionFallback;
         }
 
+        public static bool NotZero(this ReadOnlySpan<byte> x)
+        {
+            int i;
+            for (i = 0; i + 7 < x.Length; i += 8)
+            {
+                if ((x[i] | x[i + 1] | x[i + 2] | x[i + 3] | x[i + 4] | x[i + 5] | x[i + 6] | x[i + 7]) != 0)
+                    return true;
+            }
+
+            for (; i < x.Length; i++)
+            {
+                if (x[i] != 0) return true;
+            }
```

Comment:
There's a difference in the existing approach comparing to the old code: new code starts comparison from the head of `x`, and the old code started comparison from the tail of `x`. `GetBoolean` is kind of a hot-path code in the VM, thus it would be interesting to see how this bugfix affects the time of DB restore from dump.

Reply:
The old code started comparison from the head of `x` too.

Reply:
Wouldn't this work?
```csharp
foreach (var b in Span)
    if (b != 0) return true;
```

Reply:
> There's a difference in the existing approach comparing to the old code: new code starts comparison from the head of `x`, and the old code started comparison from the tail of `x`. `GetBoolean` is kind of a hot-path code in the VM, thus it would be interesting to see how this bugfix affects the performance (time) of DB restore from dump for mainnet/testnet.



> There's a difference in the existing approach comparing to the old code: new code starts comparison from the head of `x`, and the old code started comparison from the tail of `x`. `GetBoolean` is kind of a hot-path code in the VM, thus it would be interesting to see how this bugfix affects the performance (time) of DB restore from dump for mainnet/testnet.

Now the performance has been optimized(use `SequenceEqual`) and is even better than before.
```
// * Summary *

BenchmarkDotNet v0.14.0, Windows 11 (10.0.22631.4037/23H2/2023Update/SunValley3)
AMD Ryzen 9 7940HS w/ Radeon 780M Graphics, 1 CPU, 16 logical and 8 physical cores
.NET SDK 8.0.401
  [Host]     : .NET 8.0.8 (8.0.824.36612), X64 RyuJIT AVX-512F+CD+BW+DQ+VL+VBMI
  DefaultJob : .NET 8.0.8 (8.0.824.36612), X64 RyuJIT AVX-512F+CD+BW+DQ+VL+VBMI


| Method        | length | Mean     | Error     | StdDev    |
|-------------- |------- |---------:|----------:|----------:|
| LongPointer   | 7      | 2.445 ns | 0.0711 ns | 0.0665 ns |
| SequenceEqual | 7      | 2.265 ns | 0.0649 ns | 0.0722 ns |
| LongPointer   | 10     | 1.121 ns | 0.0028 ns | 0.0026 ns |
| SequenceEqual | 10     | 1.122 ns | 0.0059 ns | 0.0050 ns |
| LongPointer   | 15     | 2.430 ns | 0.0422 ns | 0.0374 ns |
| SequenceEqual | 15     | 2.342 ns | 0.0411 ns | 0.0364 ns |
| LongPointer   | 23     | 3.343 ns | 0.0181 ns | 0.0169 ns |
| SequenceEqual | 23     | 3.352 ns | 0.0086 ns | 0.0071 ns |
| LongPointer   | 32     | 3.630 ns | 0.0067 ns | 0.0059 ns |
| SequenceEqual | 32     | 3.633 ns | 0.0144 ns | 0.0135 ns |
```